other/no-loss-lottery/staking/src/lib.rs

```diff
@@ -0,0 +1,185 @@
+use scrypto::prelude::*;
+
+blueprint! {
+    struct Staking {
+        // staking and rewards pools
+        staking_pool: Vault,
+        rewards_pool: Vault,
+        // reward value by staking pool amount
+        reward_value: Decimal,
+        // last update epoch
+        last_update: u64,
+        // reward rate per staking
+        rate: Decimal,
+        // collects rewards per account
+        rewards: HashMap<Address, Decimal>,
```

Comment:
Now that we have NFTs, you could create a NFT instead of a badge when users register and store the rewards, balances and paid amount inside its metadata.